alacritty_terminal/src/term/mod.rs

```diff
@@ -927,13 +936,13 @@ impl<T> Term<T> {
         self.damage.damage_point(point);
     }
 
-    /// Damage `Vi` mode cursor.
     #[inline]
-    pub fn damage_vi_cursor(&mut self) {
-        let line = (self.grid.display_offset() as i32 + self.vi_mode_cursor.point.line.0)
-            .clamp(0, self.screen_lines() as i32 - 1) as usize;
-        let vi_point = Point::new(line, self.vi_mode_cursor.point.column);
-        self.damage.damage_point(vi_point);
+    fn grid_point_to_viewport(&self, point: Point) -> Point<usize> {
+        let display_offset = self.grid.display_offset() as i32;
+        let lines = self.screen_lines() as i32 - 1;
+        let line = (display_offset + point.line.0).clamp(0, lines) as usize;
+
+        Point::new(line, point.column)
     }
```

Comment:
If this method was made public, we could probably use it in the Alacritty crate, right? I think there are a couple instances where we use something very similar. Though those return an `Option` I think, which is something that should be considered here too since technically there's nothing in this method saying the point has to be within the viewport.

Reply:
I think we had such methods in the past and they always got removed.

Anyway, I've moved those methods from alacritty into alacritty_terminal. Not sure that we should add clamping to bounds...